File: src/android/AxiomMobileTrust.java
package com.mollatech.cordova.plugin;

import org.apache.cordova.CallbackContext;
import org.apache.cordova.CordovaInterface;
import org.apache.cordova.CordovaPlugin;
import org.apache.cordova.CordovaWebView;
import org.json.JSONArray;
import org.json.JSONObject;

import com.bluebricks.axiom.mobiletrust.app.helper.AxiomHelper;
import com.mollatech.axiom.mobiletrust.face.AxiomOTPResponse;

import android.content.Context;

public class AxiomMobileTrust extends CordovaPlugin {
	private static com.mollatech.mobiletrust.face.action.MobilityTrustVaultImpl mv = null;
	private static com.mollatech.mobiletrust.face.action.MobilityTrustKickStartImpl m = null;

	final String initialize = "initialize";
	final String load = "load";
	final String unLoad = "unLoad";
	final String confirm = "confirm";
	final String update = "update";
	final String ResetPin = "ResetPin";
	final String ChangePin = "ChangePin";
	final String ConfirmResetOrChangePin = "ConfirmResetOrChangePin";
	final String enableORDisableDebug = "enableORDisableDebug";
	final String SelfDestroy = "SelfDestroy";
	final String getLogs = "getLogs";
	final String checkstatus = "checkstatus";
	final String GetOTPPlus = "GetOTPPlus";
	final String GetSOTPPlus = "GetSOTPPlus";
	final String UnlockToken = "UnlockToken";

	JSONObject jsonResponse;
	String pin = null;
	String userid = null;
	String licensekey = null;
	String regcode = null;
	String timestamp = null;
	AxiomOTPResponse otpResponse;
	Context con;
	String result = "";

	public AxiomMobileTrust() {

	}

	@Override
	public void initialize(CordovaInterface cordova, CordovaWebView webView) {
		super.initialize(cordova, webView);
		Context context = cordova.getActivity().getApplicationContext();
		con = context;
		if (mv == null) {
			mv = new com.mollatech.mobiletrust.face.action.MobilityTrustVaultImpl();
			mv.SetGPSCoordinates("73.9287829", "18.5457378");
			mv.enableORDisableDebug(true);

		}
		if (m == null) {
			m = new com.mollatech.mobiletrust.face.action.MobilityTrustKickStartImpl();
			m.loadDelgate(context);
			m.enableORDisableDebug(true);
		}
	}

	public synchronized void initialize(final JSONArray args, final CallbackContext callbackContext) {
		Runnable runnable = new Runnable() {
			public void run() {
				try {
					regcode = args.getString(0);
					userid = args.getString(1);
					pin = args.getString(2);
					licensekey = args.getString(3);
					String strInitData = m.initialize(con, licensekey, userid, regcode, pin);
					jsonResponse = new JSONObject();
					jsonResponse.put("success", "true");
					jsonResponse.put("licensekey", licensekey);
					jsonResponse.put("initData", strInitData);
					callbackContext.success(jsonResponse);
				} catch (Exception e) {
					e.printStackTrace();
					callbackContext.error("Unable to Activate Token !!!");
				}

			};
		};

		this.cordova.getThreadPool().execute(runnable);
	}

	public synchronized void load(final JSONArray args, final CallbackContext callbackContext) {
		Runnable runnable = new Runnable() {
			public void run() {
				try {
					userid = args.getString(0);
					pin = args.getString(1);
					licensekey = args.getString(2);
					int res = mv.load(con, licensekey, userid, pin);
					jsonResponse = new JSONObject();
					jsonResponse.put("Response", res);
					jsonResponse.put("success", "true");
					callbackContext.success(jsonResponse);
				} catch (Exception e1) {
					callbackContext.error(e1.getMessage());
				}
			};
		};
		this.cordova.getThreadPool().execute(runnable);
	}

	public synchronized void unload(final JSONArray args, final CallbackContext callbackContext) {
		Runnable runnable = new Runnable() {
			public void run() {
				try {
					userid = args.getString(0);
					int res = mv.unLoad(userid);
					jsonResponse = new JSONObject();
					jsonResponse.put("Response", res);
					jsonResponse.put("success", "true");
					callbackContext.success(jsonResponse);
				} catch (Exception e1) {
					callbackContext.error(e1.getMessage());
				}
			};
		};
		this.cordova.getThreadPool().execute(runnable);
	}

	public synchronized void confirm(final JSONArray args, final CallbackContext callbackContext) {
		Runnable runnable = new Runnable() {
			public void run() {
				try {
					// con = MainActivity.context;
					licensekey = args.getString(0);
					userid = args.getString(1);
					pin = args.getString(2);
					String toconfirm = args.getString(3);
					int res = m.confirm(con, licensekey, userid, pin, toconfirm);
					jsonResponse = new JSONObject();
					jsonResponse.put("Response", res);
					jsonResponse.put("success", "true");
					callbackContext.success(jsonResponse);
				} catch (Exception e1) {
					callbackContext.error(e1.getMessage());
				}

			};
		};
		this.cordova.getThreadPool().execute(runnable);
	}

	public synchronized void update(final JSONArray args, final CallbackContext callbackContext) {
		Runnable runnable = new Runnable() {
			public void run() {
				try {
					// con = MainActivity.context;
					userid = args.getString(0);
					String payload = args.getString(1);
					pin = args.getString(2);
					int res = m.Update(con, userid, payload, pin);
					jsonResponse = new JSONObject();
					jsonResponse.put("Response", res);
					jsonResponse.put("success", "true");
					callbackContext.success(jsonResponse);
				} catch (Exception e1) {
					callbackContext.error(e1.getMessage());
				}
			};
		};
		this.cordova.getThreadPool().execute(runnable);
	}

	public synchronized void resetPin(final JSONArray args, final CallbackContext callbackContext) {
		Runnable runnable = new Runnable() {
			public void run() {
				try {
					licensekey = args.getString(0);
					userid = args.getString(1);
					String doctype = args.getString(2);
					String uniqueId = args.getString(3);
					pin = args.getString(4);
					String tStamp = args.getString(4);
					String res = mv.ResetPin(con, licensekey, userid, doctype, uniqueId, pin, tStamp);
					jsonResponse = new JSONObject();
					jsonResponse.put("Response", res);
					jsonResponse.put("success", "true");
					callbackContext.success(jsonResponse);
				} catch (Exception e1) {
					callbackContext.error(e1.getMessage());
				}
			};
		};
		this.cordova.getThreadPool().execute(runnable);
	}

	public synchronized void changePin(final JSONArray args, final CallbackContext callbackContext) {
		Runnable runnable = new Runnable() {
			public void run() {
				try {
					pin = args.getString(0);
					String newpin = args.getString(1);
					String tStamp = args.getString(2);
					String res = mv.ChangePin(pin, newpin, tStamp);
					jsonResponse = new JSONObject();
					jsonResponse.put("Response", res);
					jsonResponse.put("success", "true");
					callbackContext.success(jsonResponse);
				} catch (Exception e1) {
					callbackContext.error(e1.getMessage());
				}
			};
		};
		this.cordova.getThreadPool().execute(runnable);
	}

	public synchronized void confirmResetOrChangePin(final JSONArray args, final CallbackContext callbackContext) {
		Runnable runnable = new Runnable() {
			public void run() {
				try {
					String toconfirm = args.getString(0);
					boolean res = mv.ConfirmResetOrChangePin(toconfirm);
					jsonResponse = new JSONObject();
					jsonResponse.put("Response", res);
					jsonResponse.put("success", "true");
					callbackContext.success(jsonResponse);
				} catch (Exception e1) {
					callbackContext.error(e1.getMessage());
				}
			};
		};
		this.cordova.getThreadPool().execute(runnable);
	}

	public synchronized void enableORDisableDebug(final JSONArray args, final CallbackContext callbackContext) {
		Runnable runnable = new Runnable() {
			public void run() {
				try {
					String bcheck = args.getString(0);
					boolean res = mv.enableORDisableDebug(Boolean.parseBoolean(bcheck));
					jsonResponse = new JSONObject();
					jsonResponse.put("Response", res);
					jsonResponse.put("success", "true");
					callbackContext.success(jsonResponse);
				} catch (Exception e1) {
					callbackContext.error(e1.getMessage());
				}
			};
		};
		this.cordova.getThreadPool().execute(runnable);
	}

	public synchronized void selfDestroy(final JSONArray args, final CallbackContext callbackContext) {
		Runnable runnable = new Runnable() {
			public void run() {
				try {
					userid = args.getString(0);
					pin = args.getString(1);
					int res = m.selfDestroy(con, userid, pin);
					jsonResponse = new JSONObject();
					jsonResponse.put("Response", res);
					jsonResponse.put("success", "true");
					callbackContext.success(jsonResponse);
				} catch (Exception e1) {
					callbackContext.error(e1.getMessage());
				}
			};
		};
		this.cordova.getThreadPool().execute(runnable);
	}

	public synchronized void getLogs(final JSONArray args, final CallbackContext callbackContext) {
		Runnable runnable = new Runnable() {
			public void run() {
				try {
					String res = m.getLogs(con);
					jsonResponse = new JSONObject();
					jsonResponse.put("Response", res);
					jsonResponse.put("success", "true");
					callbackContext.success(jsonResponse);
				} catch (Exception e1) {
					callbackContext.error(e1.getMessage());
				}
			};
		};
		this.cordova.getThreadPool().execute(runnable);
	}

	public synchronized void checkStatus(final JSONArray args, final CallbackContext callbackContext) {
		Runnable runnable = new Runnable() {
			public void run() {
				try {
					userid = args.getString(0);
					int res = m.CheckStatus(con, userid);
					jsonResponse = new JSONObject();
					jsonResponse.put("Response", res);
					jsonResponse.put("success", "true");
					callbackContext.success(jsonResponse);
				} catch (Exception e1) {
					callbackContext.error(e1.getMessage());
				}
			};
		};
		this.cordova.getThreadPool().execute(runnable);
	}

	public synchronized void getOTPPlus(final JSONArray args, final CallbackContext callbackContext) {
		Runnable runnable = new Runnable() {
			public void run() {
				try {
					String timestamp = args.getString(0);
					otpResponse = mv.GetOTPPlus(timestamp);
					jsonResponse.put("otp", otpResponse.otp);
					jsonResponse.put("success", "true");
					callbackContext.success(jsonResponse);
				} catch (Exception e1) {
					callbackContext.error(e1.getMessage());
				}
			};
		};
		this.cordova.getThreadPool().execute(runnable);
	}

	public synchronized void getSOTPPlus(final JSONArray args, final CallbackContext callbackContext) {
		Runnable runnable = new Runnable() {
			public void run() {
				JSONObject jsonResponse = new JSONObject();
				try {
					timestamp = args.getString(0);
					String[] data = args.getString(1).split(",");
					otpResponse = mv.GetSOTPPlus(timestamp, data);
					jsonResponse.put("sotp", otpResponse.otp);
					jsonResponse.put("success", "true");
					callbackContext.success(jsonResponse);
				} catch (Exception e1) {
					callbackContext.error(e1.getMessage());
				}
			};
		};
		this.cordova.getThreadPool().execute(runnable);
	}

	public synchronized void unlockToken(final JSONArray args, final CallbackContext callbackContext) {
		Runnable runnable = new Runnable() {
			public void run() {
				try {
					userid = args.getString(0);
					licensekey = args.getString(1);
					pin = args.getString(2);
					String puk = args.getString(3);
					String res = mv.unlockToken(con, licensekey, userid, puk, pin);
					jsonResponse = new JSONObject();
					jsonResponse.put("Response", res);
					jsonResponse.put("success", "true");
					callbackContext.success(jsonResponse);
				} catch (Exception e1) {
					callbackContext.error(e1.getMessage());
				}
			};
		};
		this.cordova.getThreadPool().execute(runnable);
	}

	@Override
	public boolean execute(String action, JSONArray args, CallbackContext callbackContext) {
		// this.callbackContext = callbackContext;
		AxiomHelper axiom = new AxiomHelper();
		// con = MainActivity.context;
		axiom.SetContext(con);
		// initialize
		if (initialize.equalsIgnoreCase(action)) {
			initialize(args, callbackContext);

			// load
		} else if (load.equalsIgnoreCase(action)) {
			load(args, callbackContext);

			// unLoad
		} else if (unLoad.equalsIgnoreCase(action)) {
			unload(args, callbackContext);

			// confirm
		} else if (confirm.equalsIgnoreCase(action)) {
			confirm(args, callbackContext);

			// update
		} else if (update.equalsIgnoreCase(action)) {
			update(args, callbackContext);

			// ResetPin
		} else if (ResetPin.equalsIgnoreCase(action)) {
			resetPin(args, callbackContext);

			// ChangePin
		} else if (ChangePin.equalsIgnoreCase(action)) {
			changePin(args, callbackContext);

			// ConfirmResetOrChangePin
		} else if (ConfirmResetOrChangePin.equalsIgnoreCase(action)) {
			confirmResetOrChangePin(args, callbackContext);

			// enableORDisableDebug
		} else if (enableORDisableDebug.equalsIgnoreCase(action)) {
			enableORDisableDebug(args, callbackContext);

			// SelfDestroy
		} else if (SelfDestroy.equalsIgnoreCase(action)) {
			selfDestroy(args, callbackContext);

			// getLogs
		} else if (getLogs.equalsIgnoreCase(action)) {
			getLogs(args, callbackContext);

			// checkstatus
		} else if (checkstatus.equalsIgnoreCase(action)) {
			checkStatus(args, callbackContext);

			// GetOTPPlus
		} else if (GetOTPPlus.equalsIgnoreCase(action)) {
			getOTPPlus(args, callbackContext);

			// GetSOTPPlus
		} else if (GetSOTPPlus.equalsIgnoreCase(action)) {
			getSOTPPlus(args, callbackContext);

			// UnlockToken
		} else if (UnlockToken.equalsIgnoreCase(action)) {
			unlockToken(args, callbackContext);

		}
		return true;
	}

}

File: src/ios/AxiomMobileTrust.m
//
//  AxiomMobileTrust.m
//  MobileToken
//
//  Created by vikram sareen on 19/02/16.
//
//

#import "AxiomMobileTrust.h"
#import "MobilityTrustKickStartImpl.h"
#import "MobilityTrustVaultImpl.h"
#import "AxiomException.h"
#import "AxiomOTPResponse.h"
#import "AxiomPKIResponse.h"
#import "MobilityTrustKickStart.h"
#import "MobilityTrustVault.h"
#import "Axiom.h"


static NSString *g_push_message;
static MobilityTrustVaultImpl *mv;


@implementation AxiomMobileTrust{
    
    AxiomMobileTrust *sayHello;
    Axiom *axiom;
    NSString *userid;
    NSString *urlToActivate;
    NSString *pin;
    NSString *regcode;
    NSString *lattitude;
    NSString *longitude;
    NSString *licensekey;
    NSDictionary *jsonObj;
    CDVPluginResult *pluginResult;
    MobilityTrustKickStart *m ;
    MobilityTrustVaultImpl *mv;

    
    NSString *initialize;
    NSString *checkstatus;
    NSString *confirm;
    NSString *load ;
    NSString *update ;
    NSString *unLoad;
    NSString *ResetPin;
    NSString *ChangePin ;
    NSString *ConfirmResetOrChangePin ;
    NSString *enableORDisableDebug ;
    NSString *SelfDestroy ;
    NSString *getLogs ;
    NSString *GetOTPPlus;
    NSString *GetSOTPPlus;
    NSString *UnlockToken;
}

-(id)init{
    
    self = [super init];
    if(self)
    {
       self->initialize = @"initialize";
        self->checkstatus = @"checkstatus";
        self->confirm = @"confirm";
        self->load = @"load";
        self->update=@"update";
        self->unLoad=@"unLoad";
        self->ResetPin = @"ResetPin";
        self->ChangePin = @"ChangePin";
        self->ConfirmResetOrChangePin = @"ConfirmResetOrChangePin";
        self->enableORDisableDebug = @"enableORDisableDebug";
        self->SelfDestroy = @"SelfDestroy";
        self->getLogs = @"getLogs";
        self->GetOTPPlus = @"GetOTPPlus";
         self->GetSOTPPlus = @"GetSOTPPlus";
        self->UnlockToken = @"UnlockToken";
        
    }
    return self;
}

// initialize
-(void)initialize:(CDVInvokedUrlCommand*)command {
    regcode=[command.arguments objectAtIndex:0];
    userid=[command.arguments objectAtIndex:1];
    pin=[command.arguments objectAtIndex:2];
    licensekey=[command.arguments objectAtIndex:3];
     m = [[MobilityTrustKickStart alloc]init];
     mv=[[MobilityTrustVaultImpl alloc]init];
    [m LoadDelegates];
    [mv SetGPSCoordinates:@"101.6869" usingWith:@"3.1390"];
    axiom=[[Axiom alloc]init];
    NSLog(@"regcode   %@",regcode);
    NSLog(@" userid  %@",userid);
    NSLog(@" pin  %@",pin);
    NSLog(@" licensekey  %@",licensekey);
    
    NSLog(@" longitude  %@",longitude);
    
    NSLog(@" lattitude  %@",lattitude);
    @try {
               NSString *strInitData =[m initialize:licensekey usingWith: userid usingWith: regcode usingWith:pin];
               jsonObj=[[NSDictionary alloc]initWithObjectsAndKeys:
                 strInitData,@"initData",
                 @"true",@"success",
                licensekey,@"licensekey",
                 nil];
        pluginResult=[CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsDictionary:jsonObj];
        }
    @catch(NSException *exception){
        NSLog(@"%@",exception.reason);
        jsonObj=[[NSDictionary alloc]initWithObjectsAndKeys:
                 exception.reason,@"error",
                 @"false",@"success",
                 nil];
        pluginResult=[CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsDictionary:jsonObj];
    }
    [self.commandDelegate sendPluginResult:pluginResult callbackId: command.callbackId];
}


// load
-(void) load:(CDVInvokedUrlCommand*)command {
    userid=[command.arguments objectAtIndex:0];
    pin=[command.arguments objectAtIndex:1];
    licensekey=[command.arguments objectAtIndex:2];
    
    NSLog(@"load start userid %@", userid);
    NSLog(@"load start pin %@", pin);
    NSLog(@"load start licensekey %@", licensekey);
    [m LoadDelegates];
    [mv LoadDelegates];
    @try{
        mv = [[MobilityTrustVaultImpl alloc]init];
        m = [[MobilityTrustKickStart alloc]init];
        [mv SetGPSCoordinates:@"101.6869" usingWith:@"3.1390"];
        axiom=[[Axiom alloc]init];
        int res = [mv load:licensekey usingWith:userid usingWith: pin];
        NSNumber *ress=@(res);
        jsonObj=[[NSDictionary alloc]initWithObjectsAndKeys:
                 ress,@"Response",
                 @"true",@"success",
                 nil];
        pluginResult=[CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsDictionary:jsonObj];
    }
    @catch (NSException *exception) {
        NSLog(@"%@", exception.reason);
        jsonObj=[[NSDictionary alloc]initWithObjectsAndKeys:
                 exception.reason,@"error",
                 @"false",@"success",
                 nil];
        pluginResult=[CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsDictionary:jsonObj];
    }
    [self.commandDelegate sendPluginResult:pluginResult callbackId: command.callbackId];
}

// unLoad
- (void) unLoad:(CDVInvokedUrlCommand*)command {
    userid=[command.arguments objectAtIndex:0];
    @try{
       int res=[mv unLoad: userid];
        NSNumber *ress=@(res);
        jsonObj=[[NSDictionary alloc]initWithObjectsAndKeys:
                 ress,@"Response",
                 @"true",@"success",
                 nil];
        pluginResult=[CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsDictionary:jsonObj];

    }
    @catch (NSException *exception) {
        NSLog(@"%@", exception.reason);
        jsonObj=[[NSDictionary alloc]initWithObjectsAndKeys:
                 exception.reason,@"error",
                 @"false",@"success",
                 nil];
        pluginResult=[CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsDictionary:jsonObj];
    }
    [self.commandDelegate sendPluginResult:pluginResult callbackId: command.callbackId];
}

// confirm
- (void) confirm:(CDVInvokedUrlCommand*)command {
    licensekey=[command.arguments objectAtIndex:0];
    userid=[command.arguments objectAtIndex:1];
    pin=[command.arguments objectAtIndex:2];
    NSString *toconfirm=[command.arguments objectAtIndex:3];
    
    NSLog(@"confirm licensekey %@", licensekey );
    NSLog(@"confirm userid %@", userid );
    NSLog(@"confirm pin %@", pin );
    NSLog(@"confirm toconfirm %@", toconfirm );

    @try{
        int res=[m confirm: licensekey usingWith: userid usingWith: pin usingWith: toconfirm];
        NSNumber *ress=@(res);
        jsonObj=[[NSDictionary alloc]initWithObjectsAndKeys:
                 ress,@"Response",
                 @"true",@"success",
                 nil];
        pluginResult=[CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsDictionary:jsonObj];

    }
    @catch (NSException *exception) {
        NSLog(@"%@", exception.reason);
        jsonObj=[[NSDictionary alloc]initWithObjectsAndKeys:
                 exception.reason,@"error",
                 @"false",@"success",
                 nil];
        pluginResult=[CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsDictionary:jsonObj];
    }
     [self.commandDelegate sendPluginResult:pluginResult callbackId: command.callbackId];
}
// update
- (void) update:(CDVInvokedUrlCommand*)command {
    userid=[command.arguments objectAtIndex:0];
    NSString *payload=[command.arguments objectAtIndex:1];
    pin=[command.arguments objectAtIndex:2];
    
    NSLog(@"update start userid %@",userid);
    NSLog(@"payload %@",payload);
    NSLog(@"pin %@",pin);
    
    @try{
       int res=[m Update: userid usingWith: payload usingWith: pin];
        NSNumber *ress=@(res);
        jsonObj=[[NSDictionary alloc]initWithObjectsAndKeys:
                 ress,@"Response",
                 @"true",@"success",
                 nil];
    }
    @catch (NSException *exception) {
        NSLog(@"%@", exception.reason);
        jsonObj=[[NSDictionary alloc]initWithObjectsAndKeys:
                 exception.reason,@"error",
                 @"false",@"success",
                 nil];
        pluginResult=[CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsDictionary:jsonObj];
    }
     [self.commandDelegate sendPluginResult:pluginResult callbackId: command.callbackId];
}

// ResetPin
- (void) ResetPin:(CDVInvokedUrlCommand*)command {
    licensekey=[command.arguments objectAtIndex:0];
    userid=[command.arguments objectAtIndex:1];
    NSString *doctype=[command.arguments objectAtIndex:2];
    NSString *uniqueId=[command.arguments objectAtIndex:3];
    NSString *newpin=[command.arguments objectAtIndex:4];
    NSString *tStamp=[command.arguments objectAtIndex:5];
    @try{
        NSString *res=[mv ResetPin: licensekey usingWith: userid usingWith: doctype usingWith: uniqueId usingWith: newpin usingWith: tStamp];
        jsonObj=[[NSDictionary alloc]initWithObjectsAndKeys:
                 res,@"Response",
                 @"true",@"success",
                 nil];

    }
    @catch (NSException *exception) {
        NSLog(@"%@", exception.reason);
        jsonObj=[[NSDictionary alloc]initWithObjectsAndKeys:
                 exception.reason,@"error",
                 @"false",@"success",
                 nil];
        pluginResult=[CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsDictionary:jsonObj];
    }
}

// ChangePin
-(void) ChangePin:(CDVInvokedUrlCommand*)command{
    pin=[command.arguments objectAtIndex:0];
    NSString *newpin=[command.arguments objectAtIndex:1];
    NSString *tStamp=[command.arguments objectAtIndex:2];
    
    NSLog(@"ChangePin pin %@", pin);
     NSLog(@"ChangePin newpin %@", newpin);
     NSLog(@"ChangePin tStamp %@", tStamp);
    @try{
        NSString *res=[mv ChangePin: pin usingWith: newpin usingWith: tStamp];
        jsonObj=[[NSDictionary alloc]initWithObjectsAndKeys:
                 res,@"Response",
                 @"true",@"success",
                 nil];
        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsDictionary:jsonObj];
    }
    @catch (NSException *exception) {
        NSLog(@"%@", exception.reason);
        jsonObj=[[NSDictionary alloc]initWithObjectsAndKeys:
                 exception.reason,@"error",
                 @"false",@"success",
                 nil];
        pluginResult=[CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsDictionary:jsonObj];
    }
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}

//ConfirmResetOrChangePin
-(void) ConfirmResetOrChangePin:(CDVInvokedUrlCommand*)command{
    NSString *toconfirm=[command.arguments objectAtIndex:0];
    NSLog(@"ConfirmResetOrChangePin toconfirm %@", toconfirm);
    @try{
        [mv LoadDelegates];
        [m LoadDelegates];
        BOOL res1=[mv ConfirmResetOrChangePin: toconfirm];
        [m getLogs];
        
        NSNumber *ress=@(res1);
        
        NSLog(@"%@", ress);
        
        jsonObj=[[NSDictionary alloc]initWithObjectsAndKeys:
                 ress,@"Response",
                 @"true",@"success",
                 nil];
        pluginResult=[CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsDictionary:jsonObj];
    }
    @catch (NSException *exception) {
         NSLog(@"error");
        NSLog(@"%@", exception.reason);
        jsonObj=[[NSDictionary alloc]initWithObjectsAndKeys:
                 exception.reason,@"error",
                 @"false",@"success",
                 nil];
        pluginResult=[CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsDictionary:jsonObj];
    }
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}
//enableORDisableDebug
-(void) enableORDisableDebug: (CDVInvokedUrlCommand*)command{
    //mv = [[MobilityTrustVaultImpl alloc]init];
    NSString *bcheck=[command.arguments objectAtIndex:0];
    @try{
        BOOL res1=[mv enableORDisableDebug: bcheck];
        NSNumber *ress=@(res1);
        jsonObj=[[NSDictionary alloc]initWithObjectsAndKeys:
                 ress,@"Response",
                 @"true",@"success",
                 nil];
        pluginResult=[CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsDictionary:jsonObj];
    }
    @catch (NSException *exception) {
        NSLog(@"%@", exception.reason);
        jsonObj=[[NSDictionary alloc]initWithObjectsAndKeys:
                 exception.reason,@"error",
                 @"false",@"success",
                 nil];
        pluginResult=[CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsDictionary:jsonObj];
    }
     [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}

//SelfDestroy
-(void)SelfDestroy:(CDVInvokedUrlCommand*)command{
    m = [[MobilityTrustKickStart alloc]init];
    userid=[command.arguments objectAtIndex:0];
     pin=[command.arguments objectAtIndex:1];
    @try{
        int res=[m selfDestroy: userid usingWith: pin];
        NSNumber *res1=@(res);
        jsonObj=[[NSDictionary alloc]initWithObjectsAndKeys:
                 res1,@"Response",
                 @"true",@"success",
                 nil];
    }
    @catch (NSException *exception) {
        NSLog(@"%@", exception.reason);
        jsonObj=[[NSDictionary alloc]initWithObjectsAndKeys:
                 exception.reason,@"error",
                 @"false",@"success",
                 nil];
        pluginResult=[CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsDictionary:jsonObj];
    }
}
//getLogs
-(void)getLogs:(CDVInvokedUrlCommand*)command{
    m = [[MobilityTrustKickStart alloc]init];
    @try{
        NSString *logs=[m getLogs];
        jsonObj=[[NSDictionary alloc]initWithObjectsAndKeys:
                 logs,@"Response",
                 @"true",@"success",
                 nil];
    }@catch (NSException *exception) {
        NSLog(@"%@", exception.reason);
        jsonObj=[[NSDictionary alloc]initWithObjectsAndKeys:
                 exception.reason,@"error",
                 @"false",@"success",
                 nil];
        pluginResult=[CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsDictionary:jsonObj];
        //        return -1;
    }

}

//checkstatus
-(void) checkstatus:(CDVInvokedUrlCommand*)command{
    mv = [[MobilityTrustVaultImpl alloc]init];
    m = [[MobilityTrustKickStart alloc]init];
    userid=[command.arguments objectAtIndex:0];
    NSLog(@"checkstatus for %@", userid);
    @try{
        int res=[m CheckStatus: userid];
        NSNumber *ress=@(res);
        jsonObj=[[NSDictionary alloc]initWithObjectsAndKeys:
                 ress,@"Response",
                 @"true",@"success",
                 nil];
        pluginResult=[CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsDictionary:jsonObj];
    }
    @catch (NSException *exception) {
        NSLog(@"%@", exception.reason);
        jsonObj=[[NSDictionary alloc]initWithObjectsAndKeys:
                 exception.reason,@"error",
                 @"false",@"success",
                 nil];
        pluginResult=[CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsDictionary:jsonObj];
    }
    
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}

//unlockToken
-(void) UnlockToken:(CDVInvokedUrlCommand*)command{
    userid=[command.arguments objectAtIndex:0];
    licensekey=[command.arguments objectAtIndex:1];
    pin=[command.arguments objectAtIndex:2];
    NSString *pukCode=[command.arguments objectAtIndex:3];
    
    NSLog(@"UnlockToken userid %@", userid);
    NSLog(@"UnlockToken licensekey %@", licensekey);
    NSLog(@"UnlockToken pukCode %@", pukCode);
    NSLog(@"UnlockToken pin %@", pin);
    @try{
  NSString *res=[mv unlockToken: licensekey usingWith: userid usingWith: pukCode usingWith: pin];
        jsonObj=[[NSDictionary alloc]initWithObjectsAndKeys:
                 res,@"Response",
                 @"true",@"success",
                 nil];
        pluginResult=[CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsDictionary:jsonObj];
    }
    @catch (NSException *exception) {
        NSLog(@"%@", exception.reason);
        jsonObj=[[NSDictionary alloc]initWithObjectsAndKeys:
                 exception.reason,@"error",
                 @"false",@"success",
                 nil];
        pluginResult=[CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsDictionary:jsonObj];
    }
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}

//GetOTPPlus
-(void)GetOTPPlus:(CDVInvokedUrlCommand*)command{
    AxiomOTPResponse *axiomotpresponse;
    NSString *timestamp = nil;
    timestamp=[command.arguments objectAtIndex:0];
    NSMutableDictionary *jsonResponse1;
           @try {
                axiomotpresponse=[mv GetOTPPlus:timestamp];
                NSString *OTP= axiomotpresponse.otp;
               jsonResponse1= [NSMutableDictionary dictionaryWithObjectsAndKeys:
                  OTP,@"otp",
                nil];
            
            pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsDictionary:jsonResponse1];
            }
            @catch (NSException *exception) {
                NSLog(@"%@",exception.reason);
                jsonObj=[[NSDictionary alloc]initWithObjectsAndKeys:
                         exception.reason,@"error",
                         @"false",@"success",
                         nil];
                pluginResult=[CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsDictionary:jsonObj];
            }
            [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}


-(void)GetSOTPPlus:(CDVInvokedUrlCommand*)command{
    AxiomOTPResponse *axiomotpresponse;
    NSString *timestamp=[command.arguments objectAtIndex:0];
    NSString *challengeAnswerStr=[command.arguments objectAtIndex:1];
    NSArray *challengeAnswer=[challengeAnswerStr componentsSeparatedByString:@","];
    NSMutableDictionary *jsonResponse;
    @try {
        axiomotpresponse=[mv GetSOTPPlus:timestamp usingWith: challengeAnswer];
        NSString *OTP= axiomotpresponse.otp;
        jsonResponse= [NSMutableDictionary dictionaryWithObjectsAndKeys:
                        OTP,@"sotp",
                        nil];
        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsDictionary:jsonResponse];
    }
    @catch (NSException *exception) {
        NSLog(@"%@",exception.reason);
        jsonObj=[[NSDictionary alloc]initWithObjectsAndKeys:
                 exception.reason,@"error",
                 @"false",@"success",
                 nil];
        pluginResult=[CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsDictionary:jsonObj];
    }
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}
@end
